Formula/openshift-cli.rb

```diff
@@ -25,39 +24,34 @@ class OpenshiftCli < Formula
 
   depends_on "coreutils" => :build
   depends_on "go" => :build
+  depends_on "heimdal" => :build
   depends_on "socat"
 
   def install
     ENV["GOPATH"] = buildpath
-    dir = buildpath/"src/github.com/openshift/origin"
+    dir = buildpath/"src/github.com/openshift/oc"
     dir.install buildpath.children - [buildpath/".brew_home"]
 
     cd dir do
-      # make target is changing in >v4.1; remove this if statement when next
-      # bumping stable version
       if build.stable?
-        system "make", "all", "WHAT=cmd/oc"
+        system "make", "cross-build-darwin-amd64", "WHAT=cmd/oc"
       else
-        system "make", "all", "WHAT=staging/src/github.com/openshift/oc/cmd/oc"
+        system "make", "cross-build-darwin-amd64", "WHAT=staging/src/github.com/openshift/oc/cmd/oc"
       end
 
-      bin.install "_output/local/bin/darwin/amd64/oc"
-
-      prefix.install_metafiles
+      bin.install "_output/bin/darwin_amd64/oc"
 
       bash_completion.install "contrib/completions/bash/oc"
       zsh_completion.install "contrib/completions/zsh/oc" => "_oc"
     end
   end
 
   test do
+    # version is reported incorrectly as v4.2.0
+    # see and watch: https://github.com/openshift/oc/issues/416
+
     version_output = shell_output("#{bin}/oc version --client 2>&1")
-    assert_match "GitTreeState:\"clean\"", version_output
-    if build.stable?
-      assert_match "GitVersion:\"v#{version}", version_output
-      assert_match stable.instance_variable_get(:@resource)
-                         .instance_variable_get(:@specs)[:revision].slice(0, 9),
-                   version_output
-    end
+
+    assert_match "Client Version: v4", version_output if build.stable?
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thank you for the elaborate hints.
According to your (and the Cookbook's) suggestions I came up with this:

```bash
oc config set-context foo
oc config get-contexts -o name
# ^ output should contain "foo"
```

This should run against a kubeconfig file in `/tmp`, so we don't touch any actual user configuration. If there's a better place to put the temporary kubeconfig, please enlighten me.